src/main/java/edu/byu/ece/rapidSmith/design/subsite/CellDesign.java

```diff
@@ -806,6 +806,35 @@ public void addVivadoConstraint(XdcConstraint constraint) {
 		}
 		vivadoConstraints.add(constraint);
 	}
+
+	/**
+	 * Creates a map from port cells to the sites the cells are constrained to be placed on according to the
+	 * imported constraints.xdc.
+	 * @return the map from port cells to their sites
+	 */
+	public Map<Cell, Site> getPortConstraintMap() {
+		Map<Cell, Site> portConstraintMap = new HashMap<>();
+		if (vivadoConstraints == null)
+			return portConstraintMap;
+
+		for (XdcConstraint constraint : vivadoConstraints) {
+			if (constraint.getPackagePinConstraint() == null)
+				continue;
+
+			// Get the port cell
+			Cell portCell = this.getCell(constraint.getPackagePinConstraint().getPortName());
+			assert(portCell.isPort());
+
+			// Get the package pin's site
+			Site site = device.getSite(constraint.getPackagePinConstraint().getPinName());
```

Comment:
Same issue as the invalid cell.  Assertion probably isn't enough here.

Reply:
I've highlighted a couple issues.  However, I want this pulled into the main branch at this point and the concerns are minor and can be addressed in a later pr.